### DISCUSSION TITLE
Command to create a raw alembic_version table

### DISCUSSION THREAD

**Discussion by kasium** [sqlalchemy/alembic#962] [Ideas]
Hey,
not sure if this use case is somehow common, but I would like to request it:

In my application, I use the alembic_version table as a synchronization point, meaning that multiple instances of the app are starting in parallel and will try to lock the alembic_version table in order to execute a migration. However, this approach only works if the alembic_version table already exists.

So, my question would be, if it would be possible to a command to generate a raw alembic_version table? It would help me a lot. Currently I use heads and stamp together, but the code doesn't make it obvious want I want to do

**Comment by zzzeek** [comment]
it's a common use case, and you're actually doing it the "right" way which means you are doing the locking to make it happen.    you are on the right track with alembic stamp, just call it with "alembic stamp base", that by defintion ensures the version table is present but wont put anything into it.

**Reply by zzzeek** [reply]
i guess that doesnt work because you dont want to stamp to the base if there are already versions present.

**Comment by zzzeek** [comment]
ok we need a command "ensure_version".   I can accept PRs that have tests for this, see the tests in tests/test_command.py for other examples.

**Reply by kasium** [reply]
I'm still surprised how fast you always answer 👍 . Let me try to open a PR 😄

**Reply by zzzeek** [reply]
I don't have a choice.  if i dont whip through all my questions as soon as they come in, a few hours later i will have 3x more

**Reply by kasium** [reply]
So my idea would be to move migration.py::_ensure_version_table to another place and then add a command calling this function. What do you think? What would be a good place?

**Reply by zzzeek** [reply]
We can rename the method in case someone wants to call it from a  lower level than a command, but for the command you can call it directly.

Just drop the function into command and it shows up in the CLI

```diff
iff --git a/alembic/command.py b/alembic/command.py
index 1e79460..50e2100 100644
--- a/alembic/command.py
+++ b/alembic/command.py
@@ -612,6 +612,26 @@ def stamp(
     ):
         script.run_env()
 
+def ensure_version(
+    config: "Config",
+    sql: bool = False,
+) -> None:
+    """put nice docs here"""
+
+    script = ScriptDirectory.from_config(config)
+
+    def do_ensure_version(rev, context):
+        context._ensure_version_table()
+        return []
+
+    with EnvironmentContext(
+        config,
+        script,
+        fn=do_ensure_version,
+        as_sql=sql,
+    ):
+        script.run_env()
+
 
 def edit(config: "Config", rev: str) -> None:
     """Edit revision script(s) using $EDITOR.
```

**Reply by kasium** [reply]
thanks!
